### ISSUE TITLE
Implement format strings (%)

### ISSUE THREAD

**Issue by seven1m** [natalie-lang/natalie#869]
While working with ChunkyPNG, I got the error:

```
lib/chunky_png/color.rb:573:in `to_hex': undefined method `%' for "#%08x":String
```

The code looks like this:

```ruby
def to_hex(color, include_alpha = true)                                                                                                                                                       
  include_alpha ? ("#%08x" % color) : ("#%06x" % [color >> 8])                                                                                                                                
end
```

**Comment by herwinw** [comment]
For completeness: This is `string/module_spec` from #217

**Comment by herwinw** [comment]
Since we've almost got the second version of this feature, I guess this case can be closed

**Comment by seven1m** [comment]
The first version was completed before I abandoned it for a rewrite. 😆 

I think we should keep this issue until all the specs are passing?

**Comment by seven1m** [comment]
[note: I deleted an earlier comment that was unclear]

This feature isn't complete. I was trying to work on the specs and realized I hated working on this code because it was so hard to follow. (My fault, given the way I set it up to start.)

I think the new code is much easier to hack on and should allow us to complete this issue soon-ish. Let's leave it open until all those `string/modulo_specs.rb` are passing.
